### PR TITLE
[16] FEATURE: Delete Account + Settings

### PR DESCRIPTION
**Pull request by zrichard22**

### CODE REVIEW

**Review by HugoTourneur — DISMISSED** [review]

File: mobile-app/components/business/DeleteAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        <Image
+          source={{
+            uri: require('@/assets/images/deleting-account.png'),
+          }}
+          width='100%'
+          height='40%'
+          resizeMode='contain'
+        />
+        <Text textAlign='justify'>
+          An email will be send with a 5-digits code to finalise the deletion of
```
Comment (Lucas-Senoville) [inline mobile-app/components/business/DeleteAccount.tsx:72]: A better translation would be : "An email will be sent with a 5-digit code to complete the deletion of your account.Some data will be retained for up to 6 months after deletion due to legal restrictions."

File: mobile-app/screens/settings/confirmDeleteAccount.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            'Content-Type': 'application/json',
+            Authorization: session?.token,
+            featureToken: params.featureToken,
+          },
+        }
+      )
+      router.navigate('/signin')
+    } catch {
+      // eslint-disable-next-line no-console
+      console.log('An error occurred')
```
Comment (Lucas-Senoville) [inline mobile-app/screens/settings/confirmDeleteAccount.tsx:29]: Maybe set an error message to inform the user that something went wrong

**Review by Lucas-Senoville — CHANGES_REQUESTED** [review]

File: mobile-app/.vscode/settings.json
[the remark this answers is not in the document]
Comment (zrichard22) [reply mobile-app/.vscode/settings.json:1]: I will add this in  .gitignore

File: backend/microservices/UserAccount/src/controllers/UserControllers.ts [hunk 3]
```diff
@@ -313,6 +314,116 @@ class UserControllers {
...
+    return ctx.json({ message: 'Account deleted' })
+  }
+
+  async removeExpiredAccounts() {
+    const date = new Date()
+
+    await prisma.user.deleteMany({
+      where: {
+        deletedAt: {
+          lt: date,
```
Comment (loicleclere) [inline backend/microservices/UserAccount/src/controllers/UserControllers.ts:422]: You forgot to add the deletion after 6 months
https://instamint-g6.atlassian.net/browse/SCRUM-74

File: backend/microservices/UserAccount/src/routes/userApiRoutes.ts [hunk 4]
```diff
@@ -52,11 +52,19 @@ const userApiRoutes = async (app: Hono) => {
...
-    '/test/:email',
+    '/deleteAccount',
     authentification,
-    checkFeatureToken(securisedFeatures.Test),
-    async (ctx: Context) => ctx.json({ email: ctx.req.param().email })
+    checkFeatureToken(securisedFeatures.DeleteAccount),
+    async (ctx: Context) => UserControllers.deleteAccount(ctx)
+  )
+
+  app.post('checkCode', authentification, async (ctx: Context) =>
```
Comment (loicleclere) [inline backend/microservices/UserAccount/src/routes/userApiRoutes.ts:66]: Hello, can you add a zValidator for email and code

File: mobile-app/components/business/DeleteAccount.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+          your account. Some data will be keep up to 6 months after the deletion
+          due to legal restriction.
+        </Text>
+        {showDigitInput ? (
+          <>
+            <Text textAlign='justify'>
+              Enter the 6-digit code you received by email
```
Comment (loicleclere) [inline mobile-app/components/business/DeleteAccount.tsx:79]: It's 5 digits and not 6

**Review by loicleclere — CHANGES_REQUESTED** [review]

**Review by loicleclere — DISMISSED** [review]
Approve

**Review by loicleclere — APPROVED** [review]
Approve

**Review by Lucas-Senoville — APPROVED** [review]

**Pull request merged**